lib/ansible/plugins/become/su.py

```diff
@@ -139,15 +139,22 @@ class BecomeModule(BecomeBase):
         '口令',
     ]
 
-    def check_password_prompt(self, b_output):
+    def check_password_prompt(self, b_output: bytes) -> bool:
         """ checks if the expected password prompt exists in b_output """
-
         prompts = self.get_option('prompt_l10n') or self.SU_PROMPT_LOCALIZATIONS
-        b_password_string = b"|".join((br'(\w+\'s )?' + to_bytes(p)) for p in prompts)
+        password_prompt_strings = "|".join(re.escape(p) for p in prompts)
```

Comment:
The reason we tried to keep this as bytes was due to the localization prompts being possible in languages not friendly to Python unicode strings (at least we had cases with PY2, I'm not sure that would still be the case in a PY3 world).

Reply:
The prompts are already strings, we're just converting them to bytes later than before.